cmd/earthly/main.go

```diff
@@ -564,13 +568,24 @@ func unhideFlagsCommands(ctx context.Context, cmds []*cli.Command) {
 
 func (app *earthlyApp) run(ctx context.Context, args []string) int {
 	defer func() {
-		if app.logbusSetup != nil {
+		if app.logstream {
 			err := app.logbusSetup.Close()
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Error(s) in logbus: %v", err)
 			}
+			if app.logstreamDebugManifestFile != "" {
+				err := app.logbusSetup.DumpManifestToFile(app.logstreamDebugManifestFile)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error dumping manifest: %v", err)
+				}
+			}
 		}
 	}()
+	app.logbus.Run().SetStart(time.Now())
+	defer func() {
+		// Just in case this is forgotten somewhere else.
+		app.logbus.Run().SetEnd(time.Now(), logstream.RunStatus_RUN_STATUS_FAILURE)
```

Comment:
nit - feel free to ignore:
I wonder if we should set the error message as well (i.e., `SetFatalError`) if end was not previously set.
It could make it more obvious we forgot to handle this case.